stattlepy/Stattleship_API.py

```diff
@@ -41,21 +46,54 @@ def ss_get_results(self,**kwargs):
                                 page = value 
                         elif str(key) == 'verbose':
                                 verbose = value
+                        elif str(key) == 'stat_type':
+                             param['type'] = value
                         else:
                                 param[key] = value
                 
-                
-                if(verbose):
-                    print 'Making Initial API Request'
+                ### initial verbose to indicate request occurring
+                if verbose:
+                    print'Making Initial API Request'
                     
-                ###NOTE WALK IS NOT YET SUPPORTED
-                        
-                tmp = self.query_api(sport, league, ep, param, version, walk, page, verbose, token )
+                ### initial query       
+                tmp, return_header = self.query_api(sport, league, ep, param, version, walk, page, verbose, token)
                
+                ### make response list
                 response = list()
                
-                response.append(tmp)       
-               
+                ### set the original first parsed 
+                response.append(tmp)
+                
+                ### walk function IN PROGRESS
+                if(walk):
+                    ### set number of pages to walk through
+                    if 'total' in return_header:
+                        total_results = float(return_header['total'])
+                        rpp = float(return_header['per-page'])
+                        pages = int(math.ceil(total_results/rpp))
+                    else:
+                        pages = 1
+                        total_results = 1
+                        
+                    ### append the results from pages 2+ to the response list
+                    if 'link' in return_header:
+                        for p in range(2,pages+1):
```

Comment:
Instead of calculating the next page based on total and current page, could you parse the link header instead?

That will have a next and previous link to the "next page".

Since you are using `requests` I think there is a built in function to do that: https://github.com/kennethreitz/requests/blob/f5dacf84468ab7e0631cc61a3f1431a32e3e143c/requests/utils.py#L580

Main reason is that we may decide to remove the total number of records because it does take a little bit of time to calculate a total count on a large set of data -- and an optimization would to exclude that total count in the header.

If we do that, this will break -- and the only want to get the next page set is via the link header.

While the R wrapper doesn't do this (though I asked them to) the Ruby gem does.

Be awesome to use the link headers if possible.